src/social_auth/backends/__init__.py

```diff
@@ -286,6 +288,11 @@ def from_session_dict(self, session_data, *args, **kwargs):
         kwargs = kwargs.copy()
         saved_kwargs = {key: ctype_to_model(val) for key, val in session_data["kwargs"].items()}
         saved_kwargs.update((key, val) for key, val in kwargs.items())
+
+        if saved_kwargs["backend"] and type(saved_kwargs["backend"]) == str:
```

Comment:
Is this meant to check whether `"backend"` has a truthy/non-empty value? Because it would raise `KeyError` if it's just absent. If the `"backend"` key is conditional, this should be
```python
if "backend" in saved_kwargs and type(saved_kwargs["backend"]) == str:
```
And in this case we can simplify to
```python
if type(saved_kwargs.get("backend")) == str:
```

Also, I'd prefer `isinstance(saved_kwargs["backend"], str)` over `type(saved_kwargs["backend"]) == str`. Shouldn't be functionally different (no one should be subclassing `str`) but a matter of style.